### DISCUSSION TITLE
Any updates on "Phone number masking"?

### DISCUSSION THREAD

**Discussion by AedenThomas** [simple-login/app#555] [Feature Requests]
I have been waiting for "Phone number masking" from SimpleLogin for so long. Any updates on the matter? Also how does SimpleLogin plan to implement it? Is it like 1 virtual number per user or unlimited virtual numbers?

**Comment by nguyenkims** [comment]
@AedenThomas this is work in progress and we hope to have an alpha version at the beginning of next year. 

The way it works is you can reserve a phone number from a pool of numbers for a certain time (1 hours, 1 day, etc) using a credit system. Premium accounts would come with some credits. During this time, all SMS sent to this number can only read by you.

Please note that this can change and the final version can be a bit different.

**Reply by AedenThomas** [reply]
Wow. Very excited. Will this feature be restricted to certain countries or something?

**Reply by nguyenkims** [reply]
At the beginning, we'll probably provide France-specific numbers first.

**Reply by ghost** [reply]
> At the beginning, we'll probably provide France-specific numbers first.

i think the best is : work with an ISP to have sms enabled only number who are locked against "number port", since the first use case is for those service who mandate (STEAM, AMAZON, APPLE, etc) a number for recovery purpose.

Many will use a feature like this with one number (to have their security response), so i think it's better to have (unique number per paid user (who is completely locked on your services (to prevent simswap), and redirect the sms on our email (to reduce at much a possible the cost of sms on Simple login (and so the price of this services)))).

Because our number is tied to our identity so the service who ask them already know our name, so the privacy purpose of this is very low.

But for these less secure service who use sms as recovery a unique number totally locked, is much more secure than our own number known by many more person, very vulnerable to a sim swap / sms interception.

**Reply by nguyenkims** [reply]
> Many will use a feature like this with one number (to have their security response), so i think it's better to have (unique number per paid user

Having a dedicated number is going to quite expensive  (>$10/month, depending on the country) and I'm not sure a lot of users are willing to pay this amount. That's why in the first iteration, we're going to rent a number temporarily (e.g. 5 minutes, 1 hour) to keep the cost low.

**Reply by AedenThomas** [reply]
@nguyenkims It's been 2 years and there hasn't been any updates regarding this. Meanwhile Firefox Relay has rolled out their version

**Reply by Timm346** [reply]
@nguyenkims,

> That's why in the first iteration, we're going to rent a number temporarily (e.g. 5 minutes, 1 hour) to keep the cost low.

I don’t think this is the only solution. Here is the service that generates UNLIMITED and UNIQUE numbers per each request.

[Here is](https://www.cloaked.app) the link to this service. You can copy their problem solving approach.

And [here is](https://youtube.com/watch?v=HnB5AhjAwSQ) a video with explanation how it works.

Hope you will find it useful!

**Reply by HyphenSam** [reply]
That is in private beta and is only available in the US. I can't seem to view their pricing (just directs me to join their waitlist) and found very little information of this product. I also don't trust that video because it looks sponsored.

**Reply by gmarkley-VI** [reply]
> @nguyenkims It's been 2 years and there hasn't been any updates regarding this. Meanwhile Firefox Relay has rolled out their version

I'll check it out. Did you try it?

**Reply by nguyenkims** [reply]
@Timm346 thanks for the video, unfortunately this 2-way technique doesn't work if you don't know which number is going to contact you. For ex, you give a phone number to a local shop and they want to reach out to you but you don't know what is their phone number to whitelist it before.

**Reply by erikkrieg** [reply]
I've heard Voice Over IP (VoIP) mentioned a few times as a possible solution for this use case. The main downside is that an Internet connection is required (mobile data can be used, though).

Has this been considered? Any nonstarters with this approach?

**Reply by ghost** [reply]
@erikkrieg unfortunately VoIP are handled different than cellular numbers. When a phone number is needed to verify identity, VoIP numbers and specifically excluded as being verification types that are usable -- it doesn't matter if VoIP is tried to be use via wifi or internet or otherwise. Yes, nonstarter unfortunately.

**Reply by erikkrieg** [reply]
@leland359 That's a bummer. Do you happen to know of any good article or other reference that delves into the details? No pressure, but if you do, I'd be very curious to take a look at it.

**Reply by ghost** [reply]
@erikkrieg Unfortunately I don't know of any such article, I just have a lot of experience of trying to do just that: to use VoIP numbers for verification, (or no number when I can), and when I really would like to use them for an account creation is when I get the feedback from the site saying I can't use a 'fake number' -- or 'VOIP number' -- or other error message. There seems to be something about the data bases being shared between cell phone companies and other groups -- take your website of choice -- and checking the number against it? I've read about it in a few places, where it's 'locked down', not sure how exactly.  Ideally, I would only use services that don't require a phone number for verification as that leaves no choice but to support cell companies as last line of security defense, instead of myself setting that defense -- like with my security key for example, but then again, most people want to offload that burden, and the internet is fighting against fake account creation and has decided to go with cell phone company verified numbers because.... my guess is you need an id to create an account with a cell phone company? Hope this helps :), but I'm speculating here in part too.

**Reply by erikkrieg** [reply]
@leland359 Thanks for the extra context.

**Comment by 002gRTEah3mRYH** [comment]
> That is in private beta and is only available in the US. I can't seem to view their pricing (just directs me to join their waitlist) and found very little information of this product. I also don't trust that video because it looks sponsored.

That video has a affiliation link with reduction, so it is

**Comment by xsetiadi** [comment]
Any new update regardless of this feature?

**Comment by gmarkley-VI** [comment]
Please let us know status 2years later

**Comment by epoch-philosophy** [comment]
I am also wondering about an update on this. Phone relay is a super compelling feature.

**Comment by Schinken6** [comment]
I would love to have such a feature because recently I get bot calls once per week. I would like to add to the discussion something like cloaked (https://www.cloaked.com/)
I like how they handle this whole phone number situation it kinda feels like SimpleLogin but with numbers. I would totally use their service but the downside is it is only available in the USA.
I have also seen that Firefox Relay "Phone mask to protect your real phone number" wants to implement such a feature but it's still on the waitlist.

**Reply by ghost** [reply]
I just checked out their site https://www.cloaked.com/ , they don't advertise zero access encryption :(. They don't advertise end-to-end encryption :(. and in their privacy policy they give your data to third party service providers :(. They don't say how the phone number masking works, so it's just like other "masking" of phone number services which there are a lot of. I might be wrong on something here, on my recursive view, please lmk :).

**Reply by Schinken6** [reply]
thanks for your comment. You have valid points but since they only support america I still have to wait anyway.
Do you have any company in mind which you could recommend for usage in europe?

**Reply by ghost** [reply]
I'm not yet heard of a company this does this service well / privately in Europe either :/.
